### PR TITLE
Update nix to v0.10.0

### PR DESCRIPTION
**Pull request by mulkieran**
This requires an explicitly higher version of backtrace and of libc.

Signed-off-by: mulhern <amulhern@redhat.com>

Resolves: #263.

### CODE REVIEW

**Comment by mulkieran** [comment]
We shouldn't merge this until we can make sure we can update in stratisd as well. So there will be another PR forthcoming.

**Comment by mulkieran** [comment]
@trgill We should expect to be able to merge this w/out passing Jenkins because I think that if we get this to pass all other Jenkins executions will fail. But, once this is merged, we'll need to update Jenkins pronto.

**Comment by trgill** [comment]
@mulkieran there are only 3 Jenkins nodes.  I can update them as soon as we merge this...

File: Cargo.toml [hunk 1]
```diff
@@ -11,8 +11,9 @@ keywords = ["Linux", "device", "mapper", "libdm", "storage"]
 license = "MPL-2.0"
 
 [dependencies]
-libc = "0.2.31"
-nix = "0.9"
+backtrace = "0.3.5"
```
Comment (ignatenkobrain) [inline Cargo.toml]: hmmm, why is this needed?

Comment (mulkieran) [reply Cargo.toml]: It may be unnecessary...I'm trying without.

**Comment by mulkieran** [comment]
@trgill I may have overcomplicated the whole thing. When you do update the Jenkins nodes, maybe all you'll need to do is ```cargo update --package nix``` and the problems that I was encountering will be resolved without a hitch.

**Comment by mulkieran** [comment]
Doesn't seem to require any code changes to stratisd. In order to not have Jenkins disabled for too long, we should delay merging this until we're about to do a new devicemapper release, I think.

**Comment by agrover** [comment]
I got Jenkins passing by adding 'git clean -fxd' before calling 'make sudo_test'. @trgill does this seem reasonable?
